cmd/contour/serve.go

```diff
@@ -511,24 +538,31 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 			Converter: converter,
 			Logger:    log.WithField("context", "serviceStatusLoadBalancerWatcher"),
 		}
-		factory := clients.NewInformerFactoryForNamespace(ctx.EnvoyServiceNamespace)
-		informerSyncList.InformOnResources(factory, dynamicServiceHandler, k8s.ServicesResources()...)
 
-		g.Add(startInformer(factory, log.WithField("context", "serviceStatusLoadBalancerWatcher")))
+		for _, r := range k8s.ServicesResources() {
```

Comment:
Might help for readability to move this into a method to clean up the serve method.